### PR TITLE
536 logs improvements

### LINKED ISSUE #536: Logs - Improvements
- [x] Add labels column
- [x] Shrink timestamp column, set max-width of labels column and message to grow to remaining space
- [ ] Add filter option in components selection to show deleted components (default to off), 
- [x] Switch to time range component
- [ ] Only show components alive during time range

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #536

### CODE REVIEW

**Comment by moshloop** [comment]
@mainawycliffe need to add an empty search results

**Comment by mainawycliffe** [comment]
@moshloop you mean where there are no query params - i.e. all logs?

**Comment by mainawycliffe** [comment]
@moshloop Yeah, let me make the changes

**Comment by moshloop** [comment]
@mainawycliffe When expanding the list of tags, the column widths change - I am expecting them to remain the same and only that single row height changes

**Comment by mainawycliffe** [comment]
@moshloop spent a while playing with this, and two possible solutions I came across is to show one item per row and having a fixed width for the for the column or a combination of both. This is because the width of the label is unknown to begin with and makes it hard to predict how this will behave for different logs.

**Comment by moshloop** [comment]
@mainawycliffe - We always want a fixed width for the columns (maybe dependant on size of screen)

**Comment by mainawycliffe** [comment]
@moshloop I have set the width to fixed for all columns, with timestamp fitting content while message and labels having equal width and also made it possible to show only one column of labels with expand button as shown below:

image
image
image

**Comment by moshloop** [comment]
@mainawycliffe This doesn't match v8 example: https://tanstack.com/table/v8/docs/examples/react/column-sizing ?

**Comment by mainawycliffe** [comment]
@moshloop still working on it, was unable to continue last night when the API went down

**Comment by mainawycliffe** [comment]
@moshloop is this the kind of resizing you had in mind? Currently, doesn't save to local storage, I am finalizing this right now.

**Comment by moshloop** [comment]
@mainawycliffe This is still not working like the example:

* changing column width should not impact the width of other columns* 
* it should start out equal width it needs a more sensible default
* the resize indicator is too much - it is probaby more suited to the hover style

**Comment by mainawycliffe** [comment]
@moshloop In order to not affect other column widths, we would the width of the table to be fluid, i.e., not 100 percent; at the moment, the column re-adjusts based on the available width (100%), but if it can grow to over 100% and shrink below 100%, then this is possible. We would start with a sensible width and then move on from there. Is this the desired behavior?

**Comment by moshloop** [comment]
@mainawycliffe The table width should be full-width of the screen, and start with sensible widths for the columns which can then be changed - Something like this:  https://tanstack.com/table/v8/docs/examples/react/full-width-resizable-table is

**Comment by mainawycliffe** [comment]
@moshloop that's what I went with
image

**Comment by moshloop** [comment]
@mainawycliffe trying to make the labels smaller, just increases the width of the table and makes it scrollable

**Comment by mainawycliffe** [comment]
@moshloop I have made the table width a little larger; now it shows scrollbars, and reduced the labels column.

**Comment by moshloop** [comment]
@mainawycliffe this is still not working right - The timestamp column is not resizable but still seems to change width based on the other columns and easily gets to be shorter than the text

**Comment by mainawycliffe** [comment]
@moshloop when the width of the table is fixed to 100%, the width of the timestamp column is resizing, despite giving it constraints for max size and min size and setting the column not autoresize, the column still autosizes. Trying to figure out a solution for this.

**Comment by mainawycliffe** [comment]
@moshloop  Is this the kind of column resize indicator you had in mind? 
image

File: src/api/services/topology.ts [hunk 1]
```diff
@@ -49,7 +50,14 @@ export const getCanaries = async (params) => {
...
 
 export const getTopologyComponents = () => {
-  return IncidentCommander.get(`/component_names`);
+  return IncidentCommander.get<TopologyComponentItem[]>(`/component_names`);
+};
+
+export const getTopologyComponentByID = async (topologyID: string) => {
+  const res = await CanaryChecker.get<TopologyComponentItem[]>(
+    `/db/component_names?id=eq.${topologyID}`
+  );
```
Comment (moshloop) [inline src/api/services/topology.ts:59]:
```suggestion
  const res = await IncidentCommander.get<TopologyComponentItem[]>(
    `/component_names?id=eq.${topologyID}`
  );
```

Comment (mainawycliffe) [reply src/api/services/topology.ts]: So, we should be using `IncidentCommander` instead for all DB requests?

**Comment by moshloop** [comment]
@mainawycliffe  Can we rebase / squash - there are too many commits to merge, we shouldn't have more than 3 - 4

**Comment by mainawycliffe** [comment]
@moshloop I have squashed the commits as much as I can, and now I have about 5 commits, In newer PRs, I should get better results by squashing regularly when everything is flesh in my mind.

**Pull request merged**